crypto/modes/wrap128.c

```diff
@@ -267,22 +267,23 @@ size_t CRYPTO_128_unwrap_pad(void *key, const unsigned char *icv,
     if ((inlen & 0x7) != 0 || inlen < 16 || inlen >= CRYPTO128_WRAP_MAX)
         return 0;
 
-    memmove(out, in, inlen);
     if (inlen == 16) {
         /*
          * Section 4.2 - special case in step 1: When n=1, the ciphertext
          * contains exactly two 64-bit blocks and they are decrypted as a
          * single AES block using AES in ECB mode: AIV | P[1] = DEC(K, C[0] |
          * C[1])
          */
-        block(out, out, key);
-        memcpy(aiv, out, 8);
+        unsigned char buff[16];
+        memcpy(buff, in, inlen);
+        block(buff, buff, key);
```

Comment:
It's not like you're obliged to perform block operation in place, you can simply `block(in, buff, key)` and skip memcpy. And in case of allocation of intermediate buffer for protected secret value, it's only appropriate to wipe it with OPENSSL_cleanse. Even if it's on stack.

Styling nit. It's considered appropriate to add an empty line after declarations, in this case after buff declaration.

Reply:
@dot-asm thanks for your review. just back from vacation and updated my change per your comments.